### ISSUE TITLE
New gateway registration flow (Console)

### ISSUE THREAD

**Issue by kschiffer** [TheThingsNetwork/lorawan-stack#3947]
#### Summary
Congruently to #3522, we also want to revisit the gateway registration flow, which is currently too verbose and has many options that are usually not relevant during registration. Additionally, we can use templates of the device repository once we add support for gateways.

#### Why do we need this?
Gateway onboarding is currently not smooth.

#### What is already there? What do you see now?
Gateway registration using the gateway registration form.

#### What is missing? What do you want to see?
- Many UX improvements
  - Remove irrelevant fields
  - Revisit use flow 
  - #3823 
- Onboarding assistance from the device repository

#### How do you propose to implement this?
This issue is currently still a bit of a stub. I will follow up timely with wireframes to make this issue actionable. Implementation will likely happen in two stages:
1. Clean up of the current onboarding without device repository
2. Addition of onboarding via device repository to provide two onboarding modes, congruently with end device onboarding

#### How do you propose to test this?
While doing so, we should also extend the gateway end-to-end tests accordingly.

#### Can you do this yourself and submit a Pull Request?
I'll follow up with the wireframes and then we can decide who picks up the implementation.

**Comment by johanstokking** [comment]
> * Clean up of the current onboarding without device repository

Let's start with this indeed.

Do you need a list of fields that are advanced, with their default values? That would eliminate already some fields that are always shown now.

**Comment by johanstokking** [comment]
Other things in scope:

- Ability to set antenna placement (indoor or outdoor) (refs https://github.com/TheThingsNetwork/lorawan-stack/issues/4324)
- Filter frequency plans (refs https://github.com/TheThingsNetwork/lorawan-stack/issues/3823, probably needs gateways in device repository)

**Comment by kschiffer** [comment]
I have finalized the new UX for gateway registration:
[220517_integrated_gateway_onboarding_01.pdf](https://github.com/TheThingsNetwork/lorawan-stack/files/8715860/220517_integrated_gateway_onboarding_01.pdf)

I've taken care to include only fields that make sense to set during onboarding and let everything else to be set in the general settings of the gateway.

The new onboarding experience combines both claiming and normal registration into one form. To be able to determine claimability via gateway EUI, we need a respective endpoint in the backend that @KrishnaIyer is working on.

This also includes planning for a new frequency plan selection experience that will show recommended plans first:
https://github.com/TheThingsNetwork/lorawan-frequency-plans/issues/51

We can now already coordinate implementation of the general restructure of the gateway onboarding.

**Comment by johanstokking** [comment]
Should we triage this then? @NicolasMrad

**Comment by kschiffer** [comment]
@mjamescompton

Here's the scaffold I would suggest for this:

<img width="1091" alt="image" src="https://user-images.githubusercontent.com/5710611/171395579-f51692fa-fb52-4bae-a9bd-8caebbac4df4.png">

Let me know if you have any questions. You can also refer to [my comment on end device onboarding](https://github.com/TheThingsNetwork/lorawan-stack/issues/4847#issuecomment-1143399424). Many notes are relevant for this as well.

Again, here's the full wireframe:
https://github.com/TheThingsNetwork/lorawan-stack/files/8715860/220517_integrated_gateway_onboarding_01.pdf

Note that there are two onboarding flows, based on whether the gateway is claimable or not (`<GatewayRegistrationFormSection />` and `<GatewayClaimFormSection />`). Please make sure to resolve any unclarities about it before starting the implementation. Thanks!

**Comment by johanstokking** [comment]
References also https://github.com/TheThingsNetwork/lorawan-devices/issues/124.

If there's anything needed that is generic for a particular gateway and model, we might want to load that from the device repository too.

**Comment by kschiffer** [comment]
The new gateway onboarding foundation (as laid out here: https://github.com/TheThingsNetwork/lorawan-stack/issues/3947#issuecomment-1143492157) was just merged and is now available in OS and Enterprise.

I'm leaving this issue open since there are still some missing bits:

- https://github.com/TheThingsNetwork/lorawan-stack/issues/3823
- Extending the device repository to include gateways, and making that available through the Console https://github.com/TheThingsNetwork/lorawan-devices/issues/124

@mjamescompton is currently working on updating the documentation accordingly.

**Comment by KrishnaIyer** [comment]
The two remaining items here depend on https://github.com/TheThingsIndustries/lorawan-stack/issues/2018. I'm removing the milestone on this until we resolve the blocker.

**Comment by kschiffer** [comment]
I would even suggest to close this and track the rest of the progress in #2018 and a new issue once this is resolved.
